### PR TITLE
fix: Enable TF-based reconciler for SQLInstances without clone source

### PR DESCRIPTION
**Pull request by jasonvigil**
Fixes https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/2706

### CODE REVIEW

**Comment by yuwenma** [comment]
For the cases we found in #2706, I think we also want to add the support in the direct SQLInstance reconciler to remove system labels, and recognize/propagate the 0 int value. But I'm good if you want to do that in separate PRs.

**Comment by jasonvigil** [comment]
> For the cases we found in #2706, I think we also want to add the support in the direct SQLInstance reconciler to remove system labels, and recognize/propagate the 0 int value. But I'm good if you want to do that in separate PRs.

Yeah, this is just a quick fix to unblock the release.

File: pkg/controller/registration/registration_controller.go [hunk 1]
```diff
@@ -248,21 +248,12 @@ func registerDefaultController(r *ReconcileRegistration, config *config.Controll
...
-			if !hasTerraformController && !hasDCLController {
-				// We're always going to use the direct reconciler
-				useDirectReconcilerPredicate = nil
-				useLegacyPredicate = nil
-			}
-
-			if (hasTerraformController || hasDCLController) && useDirectReconcilerPredicate == nil {
+			} else {
 				logger.Error(fmt.Errorf("no predicate where we have multiple controllers"), "skipping direct controller registration", "group", gvk.Group, "version", gvk.Version, "kind", gvk.Kind)
 				hasDirectController = false
```
Comment (yuwenma) [inline pkg/controller/registration/registration_controller.go:258]: Previously we decouple the CRDs and controllers and let the labels on CRD to choose the controller. It only moves from TF to direct but not vice versa (which is an important precondition).  

Now the controller decision is switched to the CR object. 

I like the reconcile gate idea overall (except that it is very challenge to figure out which reconciler is being used, that could make triaging more difficult). I think my concern is mainly that the reconcile gate has changed the concept about some basic concept about the direct registry and label based reconciliations, that is not covered by any of our guides, tools, and test.

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve

I'd like to have this PR goes in to unblock users and continue the reconcile gate discussion.

**Pull request merged**